src/formState.ts

```diff
@@ -346,13 +347,16 @@ function newObjectState<T, P = any>(
     },
 
     get readOnly(): boolean {
-      return this._readOnly || this._considerReadOnly();
+      return (
+        this._readOnly ||
+        this._considerReadOnly() ||
+        (parentState && parentState().readOnly) ||
+        parentListState?.readOnly
+      );
     },
 
     set readOnly(readOnly: boolean) {
       this._readOnly = readOnly;
-      // Use this.readOnly so we can _considerReadOnly
-      getFields(this).forEach((f) => (f.readOnly = this.readOnly));
```

Comment:
I'm removing the "calling `readOnly` on a parent _pushes down_ the state to its children", and moving to a "children `readOnly` (reactively) reach up to their parent and or that with their own readonly".

Which per the PR description is more declarative / let prone to order-of-operation issues.